### DISCUSSION TITLE
Need to convert vcert.exe command to PS version

### DISCUSSION THREAD

**Discussion by VinceThompson** [Venafi/venafips#227] [Q&A] [answered]
Morning all,

I am struggling to get the syntax correct when trying to convert the following vcert.exe command to PS module version : 

vcert.exe enroll -u "https://xxxx" -t 123 -z "\VED\Policy\Azure-Certificates\test" --cn azure.local -format pkcs12 --file "C:\Temp\azure.local"

The above command works fine but cannot get PS to work.

I don't mind using username/password if that's easier for the above command.

What I am trying to achieve in PS is connect to our Venafi Server then enrol the server to Venafi and get the cert.

Can anyone help?

Thanks,
Vince

**Comment by gdbarron** [comment, marked as answer]
Hi @VinceThompson.  There are 3 steps: connect, enroll (new request), and get (export).

1. Connecting to the server is accomplished with `New-VenafiSession`.  You can have it create an access token for you or if you already have it, as in your example, you can use `New-VenafiSession -Server 'https://xxxx' -AccessToken 123`.
2. and 3. A new request and export are 2 separate functions, but can be piped from one to the next.  `New-TppCertificate -path '\ved\policy\azure-certificates\test' -Name 'azure.local' -WorkToDoTimeout 60 -PassThru | Export-VenafiCertificate -TppFormat 'PKCS #12' -OutPath '~/Temp' -PrivateKeyPassword (ConvertTo-SecureString -string 'myPassw0rd!' -AsPlainText -Force)`

A few things to note.

- If you want to skip step 1 you can provide the access token directly with `-VenafiSession myAccessToken`.  This requires an environment variable named TPP_SERVER be set.
- There is a good amount of built-in help with various examples.  Use `help FunctionName` -full.
- VenafiPS v6, which is coming out soon, will change some of the above.  The new command would be `New-VdcCertificate -Path '\ved\policy\azure-certificates\test' -Name 'azure.local' -PassThru | Export-VdcCertificate -pkcs12 -OutPath '~/Temp' -PrivateKeyPassword 'myPassw0rd!'`. To see all the changes the PR is https://github.com/Venafi/VenafiPS/pull/220.

**Reply by gdbarron** [reply]
@VinceThompson did this work for you?
